component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java

```diff
@@ -319,15 +353,90 @@ void testToBuilder() {
                 .build();
         assertEquals("meta1,data1,meta2,data2", getSchemaFields(schemaOrigin));
         Builder builder = schemaOrigin.toBuilder();
-        builder.withEntry(newEntry("data3", Type.STRING).build());
-        builder.withEntry(newEntry("meta3", Type.STRING).withMetadata(true).build());
+        builder.withEntry(newEntry("data3", Type.STRING));
+        builder.withEntry(new SchemaImpl.EntryImpl.BuilderImpl().withName("meta3")
+                .withType(Type.STRING)
+                .withMetadata(true)
+                .build());
         final Schema schemaNew = builder.build();
         assertEquals(3, schemaNew.getMetadata().size());
         assertEquals(3, schemaNew.getEntries().size());
         assertEquals(6, schemaNew.getAllEntries().count());
         assertEquals("meta1,data1,meta2,data2,data3,meta3", getSchemaFields(schemaNew));
     }
 
+    @Test
```

Comment:
sanitizeConnectionName method is entirely defined in component-api module, so, it should be mainly test in this module.

Reply:
see above

Reply:
move it back to component-api, as it does not use Schema.Entry interface

Reply:
done